File: app/utils/code_crypto.py
import os
import base64
from Crypto.Cipher import AES
from Crypto.Util.Padding import unpad

def get_key_and_iv():
    key_b64 = os.environ.get('CODE_ENCRYPTION_KEY')
    iv_b64 = os.environ.get('CODE_ENCRYPTION_IV')
    if not key_b64 or not iv_b64:
        raise ValueError('加密密钥或IV未配置，请检查环境变量')
    key = base64.b64decode(key_b64)
    iv = base64.b64decode(iv_b64)
    if len(key) != 32 or len(iv) != 16:
        raise ValueError('密钥需32字节，IV需16字节')
    return key, iv

def decrypt_code(encrypted_code_b64: str) -> str:
    try:
        key, iv = get_key_and_iv()
        ciphertext = base64.b64decode(encrypted_code_b64)
        cipher = AES.new(key, AES.MODE_CBC, iv)
        plain = unpad(cipher.decrypt(ciphertext), AES.block_size)
        return plain.decode('utf-8')
    except Exception as e:
        raise ValueError(f'Failed to decrypt code: {e}')

File: src/utils/codeCrypto.ts
import CryptoJS from 'crypto-js'

const keyB64 = import.meta.env.VITE_CODE_ENCRYPTION_KEY
const ivB64 = import.meta.env.VITE_CODE_ENCRYPTION_IV

function getKeyAndIV() {
  if (!keyB64 || !ivB64) {
    throw new Error('加密密钥或IV未配置，请检查环境变量')
  }
  const key = CryptoJS.enc.Base64.parse(keyB64)
  const iv = CryptoJS.enc.Base64.parse(ivB64)
  return { key, iv }
}

/**
 * AES-256-CBC 加密，返回Base64字符串
 */
export function encryptCode(plainCode: string): string {
  try {
    const { key, iv } = getKeyAndIV()
    const encrypted = CryptoJS.AES.encrypt(plainCode, key, {
      iv,
      mode: CryptoJS.mode.CBC,
      padding: CryptoJS.pad.Pkcs7
    })
    return encrypted.ciphertext.toString(CryptoJS.enc.Base64)
  } catch (e) {
    throw new Error('代码加密失败: ' + (e as Error).message)
  }
}
